layout/AppMenu.js

```diff
@@ -1,50 +1,90 @@
+import getConfig from "next/config";
+import React, { useContext, useState, useEffect, useCallback } from "react";
+import AppMenuitem from "./AppMenuitem";
+import { LayoutContext } from "./context/layoutcontext";
+import { MenuProvider } from "./context/menucontext";
+import ServiceRequestDialog from "@components/serviceRequest/ServiceRequestDialog";
+import { UserContext } from '../src/context/usercontext';
+import { User } from '@supabase/supabase-js'
+import { authService } from '../src/services/authService';
+import { useRouter } from 'next/navigation';
+
+
+
 // import pawIcon from '/images/paw.svg';
 
 const AppMenu = () => {
-    const { layoutConfig } = useContext(LayoutContext);
-    const [dialog, showDialog] = useState('')
-    const onClose = () => showDialog('')
-    const contextPath = getConfig().publicRuntimeConfig.contextPath;
-    const model = [
+  const { layoutConfig } = useContext(LayoutContext);
+  const [dialog, showDialog] = useState("");
+  const onClose = () => showDialog("");
+  const { user } = useContext(UserContext);
+  const contextPath = getConfig().publicRuntimeConfig.contextPath;
+  const router = useRouter();
+  const model = [
+    {
+      items: [
+        { label: "Dashboard", icon: "pi pi-fw pi-chart-line", to: "/" },
+        { separator: true },
+        { label: "Clients", icon: "pi pi-fw pi-users", to: "/clients"},
+        { separator: true },
         {
-            label: 'Home',
-            items: [
-                { label: 'Dashboard', icon: 'pi pi-fw pi-home', to: '/' },
-                { label: 'Clients', icon: 'pi pi-fw pi-users', to: '/clients' },
-                { label: 'Animals', iconSrc: `${contextPath}/images/paw.svg`, to: '/animals' },
-                { label: 'Reports', icon: 'pi pi-fw pi-book', to: '/reports' },
-                { separator: true },
-                { 
-                  label: 'New Request', 
-                  iconSrc: `${contextPath}/images/pawClipboard.svg`, 
-                  key: 'newServiceRequest', 
-                  command: () => {if (!dialog) showDialog('newServiceRequest')} 
-                }
-            ]
-        }
-    ];
+          label: "Animals",
+          iconSrc:`${contextPath}/images/paw.svg`,
+          to: "/animals",
+          class: `animals-icon`,
+        },
+        { separator: true },
+        { label: "Reports", icon: "pi pi-fw pi-book", to: "/reports" },
+        { separator: true },
+        {
+          label: "New Request",
+          iconSrc:`${contextPath}/images/pawClipboard.svg`,
+          key: "newServiceRequest",
+          class:`new-request-icon`,
+          command: () => {
+            if (!dialog) showDialog("newServiceRequest");
+          },
+        },
+        { separator: true },
+        {
+          label: "Profile",
+          icon: "pi pi-user",
+          to: "/auth/login",
+          title: user?.email || "user",
+          command: () => {
```

Comment:
I can change it to an anon arrow function instead  - `command: () => signOut()`  I cannot just set is as `command: signOut ` because its getting  a ref error that its initializing the signout function before it is declared.